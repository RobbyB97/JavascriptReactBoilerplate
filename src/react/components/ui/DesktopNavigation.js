/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <ul className="DesktopNavigation__list">
                    {this.props.internal_links &&
                        this.props.internal_links.map((link) => (
                            <li key={link.text}>
                                <a href={link.href}>
                                    {link.text}
                                </a>
                            </li>
                        ))
                    }
                </ul>
            </section>
        );
    };
};


export default DesktopNavigation;