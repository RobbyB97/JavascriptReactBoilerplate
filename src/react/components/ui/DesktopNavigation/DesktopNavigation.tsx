import React from 'react';

import DesktopNavigationLink from './DesktopNavigationLink';
import {navigationLinksType} from '../../../data/navigationLinks'


interface Props {
    links: navigationLinksType[];
}

interface State {

}


export class DesktopNavigation extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <nav className="DesktopNavigation__nav">
                    <ul className="DesktopNavigation__list">
                        {/* Links */}
                        {this.props.links &&
                            this.props.links.map((link) => (
                                <DesktopNavigationLink 
                                    key={link.text}
                                    {...link}
                                />
                            ))
                        }                 
                    </ul>
                </nav>
            </section>
        );
    };
};


export default DesktopNavigation;