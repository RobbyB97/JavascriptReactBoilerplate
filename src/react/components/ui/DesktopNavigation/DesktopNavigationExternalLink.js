/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class DesktopNavigationExternalLink extends React.Component {
    constructor(props) {
        super(props);
    };


    render() {
        return (
            <li className="DesktopNavigationLink">
                <a
                    className="DesktopNavigationLink__link"
                    href={this.props.href}
                    target="_blank"
                    rel="noopener"
                >
                    {this.props.text}
                </a>

                {/* Sublink menu */}
                {this.props.sublinks &&
                    <ul className="DesktopNavigationLink__sublink_list">
                        {this.props.sublinks.map((sublink) => (
                            <li
                                className="DesktopNavigation__sublink_link"
                                key={sublink.text}
                            >
                                <a href={sublink.href}>
                                    {sublink.text}
                                </a>
                            </li>
                        ))

                        }
                    </ul>
                }
            </li>
        );
    };
};


export default DesktopNavigationExternalLink;