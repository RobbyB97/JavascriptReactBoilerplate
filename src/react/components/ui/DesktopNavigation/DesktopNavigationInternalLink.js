/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';


/* Component */
export class DesktopNavigationInternalLink extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <li className="DesktopNavigationLink">
                <NavLink
                    className="DesktopNavigationLink__link"
                    to={this.props.route}
                >
                    {this.props.text}
                </NavLink>

                {/* Sublink menu */}
                {this.props.sublinks &&
                    <ul
                        className="DesktopNavigationLink__sublink_list"
                    >
                        {this.props.sublinks.map((sublink) => (
                            <li
                                className="DesktopNavigationLink__sublink_link"
                                key={sublink.text}
                            >
                                <NavLink to={sublink.route}>
                                    {sublink.text}
                                </NavLink>
                            </li>
                        ))}
                    </ul>
                }
            </li>
        );
    };
};


export default DesktopNavigationInternalLink;