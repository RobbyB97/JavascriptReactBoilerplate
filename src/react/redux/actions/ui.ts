import { 
    MOBILE_NAV__TOGGLE, MOBILE_NAV__ON, MOBILE_NAV__OFF, UIActionTypes 
} from '../types/ui';


export function navMenu_Mobile__Toggle(): UIActionTypes {
    return {
        type: MOBILE_NAV__TOGGLE
    }
}

export function navMenu_Mobile__On(): UIActionTypes {
    return {
        type: MOBILE_NAV__ON
    }
}

export function navMenu_Mobile__Off(): UIActionTypes {
    return {
        type: MOBILE_NAV__OFF
    }
}