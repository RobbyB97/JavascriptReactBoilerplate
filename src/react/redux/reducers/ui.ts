import { 
    UIState, MOBILE_NAV__TOGGLE, MOBILE_NAV__ON, MOBILE_NAV__OFF, UIActionTypes 
} from '../types/ui';

const initialState: UIState = {
    navMenu_Mobile: false
};

export function uiReducer(
    state = initialState,
    action: UIActionTypes
): UIState {
    switch(action.type) {
        case MOBILE_NAV__TOGGLE:
            return {
                navMenu_Mobile: state.navMenu_Mobile ? false : true
            }

        case MOBILE_NAV__ON:
            return {
                navMenu_Mobile: true
            }

        case MOBILE_NAV__OFF:
            return {
                navMenu_Mobile: false
            }

        default:
            return state;
    }
}


/*
export default (state = initialState, action) => {
    switch(action.type) {
        case 'MOBILE_NAV__TOGGLE':
            return {
                navMenu_Mobile: state.navMenu_Mobile ? false : true
            };

        case 'MOBILE_NAV__ON':
            return {
                navMenu_Mobile: true
            };
    

        case 'MOBILE_NAV__OFF':
            return {
                navMenu_Mobile: false
            };

        default:
            return state;
    };
};
*/