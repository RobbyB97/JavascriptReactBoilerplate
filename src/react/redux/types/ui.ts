/* State types */
export interface UIState {
    navMenu_Mobile: boolean;
}


/* Actions */
export const MOBILE_NAV__TOGGLE = 'MOBILE_NAV__TOGGLE';
export const MOBILE_NAV__ON = 'MOBILE_NAV__ON';
export const MOBILE_NAV__OFF = 'MOBILE_NAV__OFF';

interface MobileNavToggleAction {
    type: typeof MOBILE_NAV__TOGGLE;
}

interface MobileNavOnAction {
    type: typeof MOBILE_NAV__ON;
}

interface MobileNavOffAction {
    type: typeof MOBILE_NAV__OFF;
}

export type UIActionTypes = MobileNavToggleAction | MobileNavOnAction | MobileNavOffAction;