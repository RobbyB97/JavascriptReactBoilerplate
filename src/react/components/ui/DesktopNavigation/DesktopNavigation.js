/* Packages */
import React from 'react';


/* Components */
import DesktopNavigationInternalLink from './DesktopNavigationInternalLink';
import DesktopNavigationExternalLink from './DesktopNavigationExternalLink';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <nav className="DesktopNavigation__nav">
                    <ul className="DesktopNavigation__list">
                        {/* Internal links */}
                        {this.props.internal_links &&
                            this.props.internal_links.map((link) => (
                                <DesktopNavigationInternalLink {...link}/>
                            ))
                        }

                        {/* External links */}
                        {this.props.external_links &&
                            this.props.external_links.map((link) => (
                                <DesktopNavigationExternalLink {...link}/>
                            ))
                        }                    
                    </ul>
                </nav>
            </section>
        );
    };
};


export default DesktopNavigation;