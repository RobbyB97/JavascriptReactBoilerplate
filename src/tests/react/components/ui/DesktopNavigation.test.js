/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {DesktopNavigation} from '../../../../react/components/ui/DesktopNavigation';


/* Test Config */
let component;
beforeEach(() => {
    component = shallow(<DesktopNavigation />);
});


/* Tests */
test('Render Desktop navigation component', () => {
    expect(component).toMatchSnapshot();
});

