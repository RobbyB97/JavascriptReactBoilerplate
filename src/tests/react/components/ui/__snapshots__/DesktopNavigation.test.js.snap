// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Desktop navigation component 1`] = `
<section
  id="DesktopNavigation"
>
  <ul
    className="DesktopNavigation__list"
  />
</section>
`;
